lib/puppet/util.rb

```diff
@@ -479,6 +487,11 @@ def uri_encode(path, opts = { :allow_fragment => false })
   end
   module_function :uri_encode
 
+  def uri_unescape(path)
+    URI.unescape(path)
```

Comment:
Does the `URI.unescape` need the `$VERBOSE` treatment too? Or maybe ruby only deprecated `escape`?

Reply:
Yes, yes it does.
```
irb(main):004:0> require 'URI'
=> true
irb(main):005:0> URI.escape('path to')
(irb):5: warning: URI.escape is obsolete
=> "path%20to"
irb(main):007:0> URI.unescape('path%20to')
(irb):7: warning: URI.unescape is obsolete
=> "path to"
```

Reply:
Updated